README.md

```diff
@@ -658,6 +723,26 @@ This proposal is loosely related to a broader set of proposals, including [opera
 
 If we had user-defined value types, then it could make sense to use them in built-in features, such as [CSS Typed OM](https://developer.mozilla.org/en-US/docs/Web/API/CSS_Typed_OM_API/Guide) or the [Temporal Proposal](https://github.com/tc39/proposal-temporal). However, this is far in the future, if it ever happens; for now, it works well to use objects for these sorts of features.
 
+## What's the relationship between this proposal's Record & Tuple and TypeScript's [Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt) & [Tuple](https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple)?
+
+Although both kinds of Records relate to Objects, and both kinds of Tuples relate to Arrays, that's about where the similarity ends.
+
+Records in Typescript are a generic utility type to represent an object taking a key type matching with a value type. They still represent objects.
+
+Likewise Tuples in Typescript are a notation to express types in an array of a limited size (starting with TypeScript 4.0 they have a [variadic form](https://github.com/microsoft/TypeScript/pull/39094)).
+
+TS Records or Tuples could end up be usable on ECMAScript Records and Tuples as well, for instance:
+
+```ts
+// potential TS Record of an ECMAScript Record
+const record: RecordRecord<string, number> = #{
+  foo: 1,
+  bar: 2,
+};
+
+// potential TS Tuple of an ECMAScript Tuple
+const tuple: #[number, string] = #[1, "foo"];
```

Comment:
I think we're trying to show Tuples with today's TypeScript, so we should drop the first type annotation # character. 

If TS had support for Tuples as types, the whole annotation would be redundant.